astro-airflow-iris/{{ cookiecutter.repo_name }}/tests/test_run.py

```diff
@@ -1,37 +1,51 @@
 """
-This module contains an example test.
-
+This module contains example tests for a Kedro project.
 Tests should be placed in ``src/tests``, in modules that mirror your
-project's structure, and in files named test_*.py. They are simply functions
-named ``test_*`` which test a unit of logic.
+project's structure, and in files named test_*.py.
 """
 from pathlib import Path
 
-import pytest
-from kedro.config import OmegaConfigLoader
-from kedro.framework.context import KedroContext
-from kedro.framework.hooks import _create_hook_manager
-
-
-@pytest.fixture
-def config_loader():
-    return OmegaConfigLoader(conf_source=str(Path.cwd()))
-
-
-@pytest.fixture
-def project_context(config_loader):
-    return KedroContext(
-        package_name="{{ cookiecutter.python_package }}",
-        project_path=Path.cwd(),
-        env="local",
-        config_loader=config_loader,
-        hook_manager=_create_hook_manager(),
-    )
-
+from kedro.framework.session import KedroSession
+from kedro.framework.startup import bootstrap_project
+from kedro.io import DataCatalog
 
 # The tests below are here for the demonstration purpose
 # and should be replaced with the ones testing the project
 # functionality
-class TestProjectContext:
-    def test_project_path(self, project_context):
-        assert project_context.project_path == Path.cwd()
+
+class TestKedroSession:
+    def test_config_loader(self):
+        bootstrap_project(Path.cwd())
+
+        with KedroSession.create(project_path=Path.cwd()) as session:
+            # Test creating a Kedro session and verifying the config loader
+            context = session.load_context()
+            config_loader = context.config_loader
+            assert str(config_loader.conf_source) == str(Path.cwd() / "conf")
+
+            base_params = config_loader.get("parameters", "base")
+            assert base_params == {}
+
+    def test_data_catalog(self):
+        bootstrap_project(Path.cwd())
+
+        # Test creating a Kedro session and verifying its data catalog
+        with KedroSession.create(project_path=Path.cwd()) as session:
+            context = session.load_context()
+            catalog = context.catalog
+            assert isinstance(catalog, DataCatalog)
+
+            # Test if parameters can be loaded
+            parameters = catalog.load("parameters")
+            assert isinstance(parameters, dict)
+
+    def test_default_runner(self):
+        bootstrap_project(Path.cwd())
+        with KedroSession.create(project_path=Path.cwd()) as session:
+            # Test if the runner exists
+            assert session.run is not None
```

Comment:
Did you mean to execute the run here? Because right now you just check the `run` method on the KedroSession class. This should be `session.run()` instead. For the assert, you can make it specific to the pipeline in the starter.

Reply:
Changed the test to a very minimal:

```py
class TestKedroRun:
    def test_kedro_run(self):
        bootstrap_project(Path.cwd())

        with KedroSession.create(project_path=Path.cwd()) as session:
            assert session.run() is not None
```

Which should work allright for the context of starters, because all of the starter templates come with example pipelines that this test can run. However, if an user creates a new kedro project from the default template and selects the test tool, it'll pull the file from the default template in the main Kedro repo, so that needs to be altered as well.

I've made the change on the main Kedro repo here: https://github.com/kedro-org/kedro/pull/4368
Also added a very tiny example pipeline otherwise the test will fail. If the user does choose to have an example pipeline or to use a starter, the example pipeline will be pulled from kedro-starters instead.

Reply:
@datajoely this is just supposed to be an example test for the current pipeline though and the example projects don't use hooks. To address that, I think we should add something in the testing docs.